### PR TITLE
Refactor map component

### PR DESCRIPTION
**Pull request by laudickson**
First few thoughts trying out a refactored more 'general' component for map

Changed from vanilla js mapbox library to react-map-gl pacakge. I've found it is WAY more react-typescript friendly and felt very little friction with the type-system 

This is pretty rough and tbh the more I played with ReactMapGL I questioned if even it's necessary to have a generic in-app Map component and just use ReactMapGL component directly when you need to - their API and documentation is pretty comprehensive and they give lots of examples that offers a lot of great patterns

https://visgl.github.io/react-map-gl/examples/geojson
https://visgl.github.io/react-map-gl/docs

Also used some things from Brad's PR https://github.com/codeforboston/mattapan-mapping/blob/b909bec69422fbd94183af1a95c88824305d8511/src/components/Map/Map.tsx v useful

This PR is in draft for now, I'll spend a couple more shower thoughts and see if I can polish this further. The 'Explore' page is playground for now to see how an instance of Map component could play out (still using not-curated-data)

### CODE REVIEW

**Comment by laudickson** [comment]
Also, I deleted package.json in this changeset incase things conflate with yarn packages (I think I had it from when originally I set up CRA)

File: src/theme/Theme.tsx [hunk 1]
```diff
@@ -42,6 +42,17 @@ export const Theme: Theme = {
...
+
+    map: {
+        street: 'mapbox://styles/mapbox/streets-v11',
+        outdoor: 'mapbox://styles/mapbox/outdoors-v11',
+        light: 'mapbox://styles/mapbox/light-v10',
+        dark: 'mapbox://styles/mapbox/dark-v10',
+        satellite: 'mapbox://styles/mapbox/satellite-v9',
+        satelliteStreet: 'mapbox://styles/mapbox/satellite-streets-v11',
+        navigationDay: 'mapbox://styles/mapbox/navigation-day-v1',
+        navigationNight: 'mapbox://styles/mapbox/navigation-night-v1',
```
Comment (laudickson) [inline src/theme/Theme.tsx:55]: Threw in map styles into the Theme and worked pretty well with intellisense

File: src/@types/map.d.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+interface MapProps {
+    data: any;
+    dataLayer?: any;
```
Comment (laudickson) [inline src/@types/map.d.ts:3]: I think of dataLayer like an object that represents the styling of data

Comment (laudickson) [reply src/@types/map.d.ts:3]: Also the `any` here is temporary (and a total copout of using the type system lol. I'll need to spend just a bit more time looking at this object structure and replace it with the `any`)

File: src/@types/map.d.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+interface MapProps {
+    data: any;
```
Comment (laudickson) [inline src/@types/map.d.ts:2]: The `any` here should be a geojson type. There exists https://www.npmjs.com/package/@types/geojson but I haven't tested or looked into it at all yet

File: src/@types/theme.d.ts [hunk 4]
```diff
@@ -7,6 +7,8 @@ interface Theme {
     fonts: Fonts;
 
     colors: Colors;
+
+    map: MapStyles;
```
Comment (laudickson) [inline src/@types/theme.d.ts:11]: yay or nay?

File: src/pages/ExplorePage.tsx [hunk 5]
```diff
@@ -1,10 +1,40 @@
...
+	const paintArray: (string|number|string[])[] = [...Array(54).keys()];
+	const arrayWithColor = paintArray.map(x => [x, generateColor()])
+	const flattened = arrayWithColor.flat();
+	flattened.unshift('match', ['get', 'FID']);
+	flattened.push('blue');
+
+	return flattened
+}
+
+const generateColor = () => '#' + (Math.random() * 0xFFFFFF << 0).toString(16).padStart(6, '0');
```
Comment (laudickson) [inline src/pages/ExplorePage.tsx:16]: This stuff is temporary, just helper stuff to support dataLayer const below

File: src/pages/ExplorePage.tsx [hunk 6]
```diff
@@ -1,10 +1,40 @@
...
+    'fill-opacity': [
+      'case',
+      ['boolean', ['feature-state', 'hover'], false],
+      .8,
+      0.5
+  ]
+  }
+};
+
+const mapData= MattapanZoningSubdistrictsData()
```
Comment (laudickson) [inline src/pages/ExplorePage.tsx:32]: Not sure if data and dataLayer should just be in this page component or passed in as a prop in the parent component (App.tsx)

Also I can't remember something something the reason to use the router to manipulate the data...? something like that, does anyone recall?

File: src/@types/map.d.ts (hunk 3 above)
Comment (thadk) [reply src/@types/map.d.ts]: I trust the way this library's @types does GeoJSON types if that's any help: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/f7ec78508c6797e42f87a4390735bc2c650a1bfd/types/topojson-client/topojson-client-tests.ts

File: src/theme/Theme.tsx (hunk 1 above)
Comment (thadk) [reply src/theme/Theme.tsx:55]: 🤩

File: src/@types/theme.d.ts (hunk 4 above)
Comment (thadk) [reply src/@types/theme.d.ts:11]: yay

**Comment by thadk** [comment]
Thanks for choosing our opinion(s) on the Mapbox library setup, @laudickson!

Not sure why the deploy preview situation is so overloaded but this is the one that built I think, though the mapbox token might not be right still https://deploy-preview-69--mattapan-mapping.netlify.app/#/
Do you happen to have 1-3 netlify's going in your account there?

**Comment by laudickson** [comment]
I'll go over and share more in-depth with group tomorrow, but I'm somewhat settling on this pattern for now -

* All Map-like components in `MapAtoms`; I don't think having `Map` in organism makes sense anymore
* Whenever you want to make a map with a geojson data-type (or multiple geojsons), the geojson you want to import has to have a corresponding 'layer' with it; basically a json that describes how the geojson will be styled (follows `LayerProp` as described by react-map-gl`

Ex. pattern loosely looks like
```
import geoJsonData1 from './geoJsonData1';
import geoJsonData2 from './geoJsonData2';

const layer1 = { id: 'foo', type: 'line', paint: { 'line-color': 'orange' } };
const layer2 = { id: 'bar', type: 'fill', paint: {fill-color': 'purple' } };

      <Map mapStyle={ Theme.map.dark }>
        <>
          <MapSource data={ geoJsonData1 } layer={ layer1 } />
          <MapSource data={ geoJsonData2 } layer={ layer2} />
        </>
      </Map>
```

Concerns/issues:
* Type casting geojson data before able to actually put into as prop value (omitted in example)
* necessary fragment brackets because React children only expects 1 child (how to get around?)
* questioning the whole pattern itself / appropriate in MapAtoms?

File: src/pages/ExplorePage.tsx [hunk 7]
```diff
@@ -1,10 +1,35 @@
...
+import { Theme } from '@/theme/Theme';
+import { GreaterMattapanZoning } from '@/data/GreaterMattapanZoning';
+import { BlueHillAveCorridor } from '@/data/BlueHillAveCorridor';
+
+const GreaterMattapanZoningData = GreaterMattapanZoning as MapData;
+const GreaterMattapanZoningLayer = {
+  id: 'greaterMattapanZoning',
+  type: 'line',
+  paint: {
+    'line-color': 'yellow',
```
Comment (laudickson) [inline src/pages/ExplorePage.tsx:11]: This could be a color as defined from color palette

**Comment by laudickson** [comment]
> Thanks for choosing our opinion(s) on the Mapbox library setup, @laudickson!
> 
> Not sure why the deploy preview situation is so overloaded but this is the one that built I think, though the mapbox token might not be right still https://deploy-preview-69--mattapan-mapping.netlify.app/#/ Do you happen to have 1-3 netlify's going in your account there?
> 
> Final build was most successful I guess: https://mattapan-mapping.netlify.app/explore

@thadk good look there, it has been working locally for me pretty much all this time but turns out it was related to this https://github.com/visgl/react-map-gl/issues/1266 and this version of mapbox isn't transpiring correctly. Lots of suggested solutions in that thread, but this one seemed to work as far as from looking at the deploy preview (craco ftw once again!)

* https://github.com/mapbox/mapbox-gl-js/issues/10173#issuecomment-793462984
* https://deploy-preview-69--mattapan-mapping.netlify.app/explore

File: src/pages/ExplorePage.tsx [hunk 8]
```diff
@@ -1,10 +1,33 @@
...
   return (
-    <>
-      <Map />
-      <Map2 />
-    </>
+    <MapContainer>
+      <Map mapStyle={ Theme.map.dark }>
+        <>
+          <MapGeoJsonSource data={ GreaterMattapanZoningData } layer={ GreaterMattapanZoningLayer } />
+          <MapGeoJsonSource data={ BlueHillAveCorridorData } layer={ BlueHillAveCorridorLayer} />
```
Comment (laudickson) [inline src/pages/ExplorePage.tsx:28]: Add palette keys

File: src/@types/map.d.ts [hunk 9]
```diff
@@ -0,0 +1,19 @@
+interface MapProps {
+    startingCoordinates?: { lat: number, lng: number };
+    mapStyle?: string;
+    children?: React.ReactChild;
+}
+
+type MapLayer = LayerProps;
+type MapData = Feature<Geometry, GeoJsonProperties> | FeatureCollection<Geometry, GeoJsonProperties>;
```
Comment (laudickson) [inline src/@types/map.d.ts:8]: rename to MapGeoJson

File: src/@types/map.d.ts [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+
+interface MapStyles {
```
Comment (laudickson) [inline src/@types/map.d.ts:10]: rename to MapBaseStyles

**Comment by laudickson** [comment]
and move assets and boundaries into public folder and use from there

**Pull request merged**